java/semantickernel-api/src/main/java/com/microsoft/semantickernel/orchestration/ToolCallBehavior.java

```diff
@@ -59,14 +59,15 @@ public UnmodifiableToolCallBehavior unmodifiableClone() {
     }
 
     /**
-     * Enable or disable kernel functions. If kernel functions are disabled, they will not be passed
-     * to the model. This is the default behavior.
+     * Enable kernel functions. If kernel functions are disabled, they will not be passed
+     * to the model unless specific functions have been enabled via {@code enableFunctions}.
+     * <p>
+     * By default, all kernel functions are disabled.
      *
-     * @param enable Whether to enable kernel functions.
      * @return This ToolCallBehavior.
      */
-    public ToolCallBehavior enableKernelFunctions(boolean enable) {
-        this.kernelFunctionsEnabled = enable;
+    public ToolCallBehavior enableKernelFunctions() {
```

Comment:
Can we update ToolCallBehaviour to construct via builder pattern and make fields final